graphs-lab/src/main/kotlin/graphs_lab/algs/DijkstraAlgorithm.kt

```diff
@@ -0,0 +1,170 @@
+package graphs_lab.algs
+
+import graphs_lab.algs.utils.checkAndGetFirst
+import graphs_lab.algs.utils.checkAndGetSecond
+import graphs_lab.core.edges.WeightedEdge
+import graphs_lab.core.graphs.WeightedGraph
+
+/**
+ * The Dijkstra algorithm.
+ *
+ * Computes the shortest paths from source vertex to all vertices in a weighted graph.
+ * The Dijkstra algorithm does not support negative edge weights and in such cases an exception is thrown.
+ * The shortest path is a path with a minimal sum of the all weights of edges on the path.
+ *
+ * @references https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm
+ *
+ * @param I the type of the vertex identifiers
+ * @property graph the input graph
+ */
+class DijkstraAlgorithm<I>(val graph: WeightedGraph<I>) {
+	/**
+	 * [table] the map that associates the identifier of a vertex with a pair that consist of
+	 * the identifier of a vertex, which came from beginning with the given identifier, and
+	 * a weight equals the sum of the all weights of edges on the path from the given identifier.
+	 * A path is a set of the identifiers of a vertex that get if add the identifiers first elements of pairs
+	 * to this set beginning with the given identifier.
+	 */
+	private val table = mutableMapOf<I, Pair<I?, Double>>()
+
+	/**
+	 * [statusTable] the status of [table] equals a pair that consist of the given identifier,
+	 * from which come out, and a boolean variable that shows whether the algorithm was
+	 * processed without exceptions.
+	 */
+	private var statusTable: Pair<I, Boolean>? = null
+
+	/**
+	 * Returns the shortest path from a source vertex to a target vertex.
+	 *
+	 * @param idSource the identifier of the source vertex
+	 * @param idTarget the identifier of the target vertex
+	 * @return the path equals a list of the identifier of a vertex in the order of movement
+	 * @throws IllegalArgumentException if a weight is negative number or [idSource] or [idTarget]
+	 * is not contained in the graph
+	 * @throws IllegalStateException if there is an appeal to a non-existent identifier
+	 */
+	fun getPath(idSource: I, idTarget: I): MutableList<I> {
```

Comment:
Прдеполагаю, что путь который запросит пользователь должен быть unmutable:
```suggestion
	fun getPath(idSource: I, idTarget: I): List<I> {
```